### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by rffontenelle**
- Add latest changes into CHANGES.rst
- Update CI status badge in README and doc/index.rst, as CI changed from Travis to GitHub Actions
- Also update development instructions with new CI solution
- Replace info regarding old Python transifex-client library with the new implementation of Transifex CLI tool

P.s.: I did not add latest CI-specific pull requests to CHANGES.rst as they were not added in previous changes before, so I'm assuming it is intentional not add them.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -7,19 +7,24 @@ CHANGES
...
+- #62: Migrate from transifex-client to transifex cli
 
 Documentation
 -------------
 
 Bug Fixes
 ---------
+- #53: Set locale of created po files
 - #55: FileNotFoundError on update-txconfig-resources in the subdirectory
+- #72: Enable build of 3.11 and 3.12
```
Comment (AA-Turner) [inline CHANGES.rst:26]:
```suggestion
- #72: Declare support for Python 3.11 and Python 3.12
```

File: CHANGES.rst [hunk 2]
```diff
@@ -7,19 +7,24 @@ CHANGES
...
+- #73: Fix issues reported by flake8
```
Comment (AA-Turner) [inline CHANGES.rst:27]:
```suggestion
```

File: doc/basic.rst [hunk 3]
```diff
@@ -13,7 +13,7 @@ Basic Features
 
 Optional features
 ==================
-These features depends to `transifex-client`_ library.
+These features depends on `transifex-client`_ tool.
```
Comment (AA-Turner) [inline doc/basic.rst:16]:
```suggestion
These features depends on the `transifex-client`_ tool.
```

File: doc/dev.rst [hunk 4]
```diff
@@ -55,7 +58,7 @@ New package version
 
 The sphinx-intl package will be uploaded to PyPI: https://pypi.org/project/sphinx-intl/.
 
-Here is a release procefure for releasing.
+Here is a release procedure for releasing.
```
Comment (AA-Turner) [inline doc/dev.rst:61]:
```suggestion
Here is a release procedure:
```

**Review by AA-Turner — COMMENTED** [review]
3.11/12 should probably be under "environments"

A

File: CHANGES.rst [hunk 5]
```diff
@@ -7,19 +7,23 @@ CHANGES
 
 Environments
 ------------
+- #71: Drop Python 3.5 support
```
Comment (AA-Turner) [inline CHANGES.rst:11]:
```suggestion
- #71: Drop Python 3.5 support
- #72: Declare support for Python 3.11 and Python 3.12

```

File: CHANGES.rst [hunk 6]
```diff
@@ -7,19 +7,23 @@ CHANGES
...
+- #62: Migrate from transifex-client to transifex cli
 
 Documentation
 -------------
 
 Bug Fixes
 ---------
+- #53: Set locale of created po files
 - #55: FileNotFoundError on update-txconfig-resources in the subdirectory
+- #72: Declare support for Python 3.11 and Python 3.12
```
Comment (AA-Turner) [inline CHANGES.rst:26]:
```suggestion
```

**Review by AA-Turner — APPROVED** [review]

**Comment by rffontenelle** [comment]
Thanks for moving it to Environments. I also starting the commit, but from smartphone is kinda slow actions.

**Pull request merged**